### PR TITLE
Implement login endpoint for Swedish BankID method

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Integrate the auth service to be able to use the "member login" endpoint.

Add a new API route to make the request to auth service.

This new API route accepts an SSN + optional "next" parameter to redirect the user after successful login. It also sets the auth session cookie.

## Justify why they are needed

This will be useful in the future but more so if we want to handle existing members in the checkout. The idea is to make them login before going through the checkout.

I added an API route beacuse CORS blocks us from calling auth service directly from client.

We don't know what the UX will look like so this PR doesn't contain any UI stuff.

I want to co-locate all auth stuff in the same "service" module -- will do in separate PR.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1318** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1318" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1318?utm_source=stack-comment).

**Comment by alebedev** [comment]
General comment - we can allow CORS on per-endpoint basis in auth server.  If this is the only reason to make request server-side, we can move it to the client

File: apps/store/src/pages/api/auth/login/se.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const ssn = req.body.ssn
+  if (typeof ssn !== 'string') throw new Error('No ssn provided')
+
+  const accessToken = await memberLoginSE(ssn)
+  LOGGER.debug('Login successful')
+
+  Auth.save(accessToken, { req, res })
+
+  const nextURL = new URL(ORIGIN_URL)
+  nextURL.pathname = PageLink.home({ locale: 'se' })
```
Comment (alebedev) [inline apps/store/src/pages/api/auth/login/se.ts:21]: Can we somehow take client locale?

File: apps/store/src/pages/api/auth/login/se.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  const nextQueryParam = req.query['next']
+  if (typeof nextQueryParam === 'string') {
```
Comment (alebedev) [inline apps/store/src/pages/api/auth/login/se.ts:24]: Is it intentional that we're ignoring any query params in next URL?

File: apps/store/src/utils/auth.ts [hunk 3]
```diff
@@ -1,32 +1,116 @@
 import { CookieParams, getAuthHeader } from '@/services/Auth/Auth'
 
+enum MemberLoginMethod {
+  SE_BANKID = 'SE_BANKID',
+}
+
+const BASE_URL = process.env.NEXT_PUBLIC_AUTH_ENDPOINT
+const Endpoint = {
```
Comment (alebedev) [inline apps/store/src/utils/auth.ts:8]: nice!

**Review by alebedev — COMMENTED** [review]
Looks good.  Only major question I have is client vs backend request

File: apps/store/src/pages/api/auth/login/se.ts (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/pages/api/auth/login/se.ts:21]: We would have to pass it in or read from a cookie I guess

Let's worry about it if we decide we need this endpoint in the first place.

File: apps/store/src/pages/api/auth/login/se.ts (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/pages/api/auth/login/se.ts:24]: What do you mean? What are we ignoring?

**Comment by robinandeer** [comment]
@alebedev yeah I figured we could enable CORS as well but just wanted to try it out with the setup we have at the moment

File: apps/store/src/pages/api/auth/login/se.ts (hunk 2 above)
Comment (alebedev) [reply apps/store/src/pages/api/auth/login/se.ts:24]: Let's say we want `?next=/path?q=1` (properly urlencoded).  The way you set pathname, you'll redirect to `/path` without any query params

**Review by alebedev — APPROVED** [review]
Looks good, did round of pair improvements

**Comment by robinandeer** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: s3qmEzbk8MIL7p5CwkCI.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1318)

**Pull request merged**